kolibri/core/assets/src/api-resources/masteryLog.js

```diff
@@ -26,4 +26,21 @@ export default new Resource({
   fetchDiff(id) {
     return this.fetchDetailModel('diff', id);
   },
+
+  /**
+   * Get's the diff of the most recent mastery log compared with the previous attempt, for
+   * this content_id and user_id pair.
+   * with annotations:
+   *  - `diff`: object or null, with `correct` and `time_spent` diffs
+   *  - `attemptlogs`: list of attempt logs with annotated diff with previous try's attempt
+   *
+   * @param {String} content_id
+   * @param {String} user_id
+   * @return {Promise}
+   */
+  fetchMostRecentDiff({ content_id, user_id } = {}) {
+    return this.getListEndpoint('most_recent_diff', { content: content_id, user: user_id }).then(
+      response => response.data
+    );
+  },
```

Comment:
Not worth changing now, but I figured since we had to load the list of tries anyway, we could call the diff endpoint for the first one (most recent) from that response. This works!

Reply:
Yeah, I did think of that, but I have been fighting against chaining requests lately to try to make things as snappy as possible.